packages/node/src/assignment/assignment-service.ts

```diff
@@ -0,0 +1,64 @@
+import { BaseEvent } from '@amplitude/analytics-types';
+import { CoreClient } from '@amplitude/analytics-types';
+import {
+  Assignment,
+  AssignmentFilter,
+  AssignmentService,
+} from 'src/assignment/assignment';
+import { hashCode } from 'src/util/hash';
+
+export const DAY_MILLIS = 24 * 60 * 60 * 1000;
+const FLAG_TYPE_MUTUAL_EXCLUSION_GROUP = 'mutual-exclusion-group';
+
+export class AmplitudeAssignmentService implements AssignmentService {
+  private readonly amplitude: CoreClient;
+  private readonly assignmentFilter: AssignmentFilter;
+
+  constructor(amplitude: CoreClient, assignmentFilter: AssignmentFilter) {
+    this.amplitude = amplitude;
+    this.assignmentFilter = assignmentFilter;
+  }
+
+  async track(assignment: Assignment): Promise<void> {
+    if (this.assignmentFilter.shouldTrack(assignment)) {
+      this.amplitude.logEvent(this.toEvent(assignment));
```

Comment:
we might as well call await on this too, the caller does not need to actually wait for the return to continue it's execution.